### PR TITLE
Cast skills given on action parameter on wave start.

### PR DESCRIPTION
**Pull request by Unengine**
### Description

1. SSIA

### How to test

1. Check if stage buff is properly casted during clearing uncleared world.

### Required Reviewers
@sonohoshi

### CODE REVIEW

File: nekoyume/Assets/_Scenes/Game.unity [hunk 1]
```diff
@@ -224,7 +224,7 @@ RectTransform:
   m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
   m_AnchorMin: {x: 0.5, y: 0.5}
   m_AnchorMax: {x: 0.5, y: 0.5}
-  m_AnchoredPosition: {x: 0, y: 0}
+  m_AnchoredPosition: {x: 0, y: 0.09}
```
Comment (boscohyun) [inline nekoyume/Assets/_Scenes/Game.unity:227]: Is this intended?

File: nekoyume/Assets/_Scripts/BlockChain/ActionHandler.cs [hunk 2]
```diff
@@ -102,6 +102,21 @@ protected GoldBalanceState GetGoldBalanceState<T>(ActionBase.ActionEvaluation<T>
...
+            var buffStateAddress = Addresses.GetBuffStateAddressFromAvatarAddress(avatarAddress);
+            if (evaluation.OutputStates.GetState(buffStateAddress) is
+                Bencodex.Types.List serialized)
+            {
+                var state = new HackAndSlashBuffState(buffStateAddress, serialized);
+                return state;
+            }
+
+            return null;
+        }
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionHandler.cs:118]: How about this?

```suggestion
        protected static UniTask<HackAndSlashBuffState> GetHackAndSlashBuffStateAsync<T>(
            ActionBase.ActionEvaluation<T> evaluation) where T : ActionBase
        {
            var avatarAddress = States.Instance.CurrentAvatarState.address;
            var buffStateAddress = Addresses.GetBuffStateAddressFromAvatarAddress(avatarAddress);
            if (await evaluation.OutputStates.GetStateAsync(buffStateAddress) is
                Bencodex.Types.List serialized)
            {
                var state = new HackAndSlashBuffState(buffStateAddress, serialized);
                return state;
            }

            return null;
        }
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionHandler.cs [hunk 3]
```diff
@@ -214,6 +229,17 @@ protected async UniTask UpdateCurrentAvatarStateAsync()
...
+            ActionBase.ActionEvaluation<T> evaluation) where T : ActionBase
+        {
+            var state = GetHackAndSlashBuffState(evaluation);
+
+            if (state is { })
+            {
+                States.Instance.SetHackAndSlashBuffState(state);
+            }
+        }
+
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionHandler.cs:242]: If the `GetHackAndSlashBuffState()` function will be async function,

```suggestion
        protected static UniTask UpdateHackAndSlashBuffStateAsync<T>(
            ActionBase.ActionEvaluation<T> evaluation) where T : ActionBase
        {
            var state = await GetHackAndSlashBuffStateAsync(evaluation);

            if (state is { })
            {
                States.Instance.SetHackAndSlashBuffState(state);
            }
        }

```

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 4]
```diff
@@ -992,6 +1002,7 @@ private void ResponseHackAndSlash(ActionBase.ActionEvaluation<HackAndSlash> eval
                             {
                                 UpdateCurrentAvatarStateAsync(eval).Forget();
                                 UpdateWeeklyArenaState(eval);
+                                UpdateHackAndSlashBuffState(eval);
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:1005]: If the `UpdateHackAndSlashBuffState()` function will be async function,

```suggestion
                                UpdateHackAndSlashBuffStateAsync(eval).Forget();
```

or

```suggestion
private UniTaskVoid ResponseHackAndSlashAsync(eval)
{
                                await UpdateHackAndSlashBuffStateAsync(eval);
}
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 5]
```diff
@@ -1441,6 +1466,32 @@ private void ResponseUnlockWorld(ActionBase.ActionEvaluation<UnlockWorld> eval)
...
+        {
+            if (!(eval.Exception is null))
+            {
+                Debug.LogError($"HackAndSlashRandomBuff exc : {eval.Exception.InnerException}");
+                return;
+            }
+
+            UpdateCurrentAvatarStateAsync(eval).Forget();
+            UpdateAgentStateAsync(eval).Forget();
+            UpdateHackAndSlashBuffState(eval);
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:1479]: If the UpdateHackAndSlashBuffState() function will be async function,

```suggestion
        private UniTaskVoid ResponseHackAndSlashRandomBuffAsync(ActionBase.ActionEvaluation<HackAndSlashRandomBuff> eval)
        {
            if (!(eval.Exception is null))
            {
                Debug.LogError($"HackAndSlashRandomBuff exc : {eval.Exception.InnerException}");
                return;
            }

            UniTask.WhenAll(
                UpdateCurrentAvatarStateAsync(eval),
                UpdateAgentStateAsync(eval),
                UpdateHackAndSlashBuffStateAsync(eval)
            );
```

or just forget that.

```
UpdateHackAndSlashBuffStateAsync(eval).Forget();
```

**Review by boscohyun — COMMENTED** [review]
The point of this review is that state should be got asynchronously to avoid freezing.

File: nekoyume/Assets/_Scenes/Game.unity (hunk 1 above)
Comment (Unengine) [reply nekoyume/Assets/_Scenes/Game.unity:227]: Not intended. I'll fix it!

File: nekoyume/Assets/_Scripts/UI/Module/Button/BonusBuffButton.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            {
+                gameObject.SetActive(false);
+                return;
+            }
+
+            _stageId = currentStageId;
+            _hasEnoughStars = skillState.StarCount >= row.MaxStar;
+            starCountText.text = $"{skillState.StarCount}/{row.MaxStar}";
+            _disposableForOnDisabled = Widget.Find<BuffBonusResultPopup>().OnBuffSelectedSubject
+                .Subscribe(_ => SetIcon(skillState))
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Module/Button/BonusBuffButton.cs:70]:
```suggestion
            _disposableForOnDisabled = Widget.Find<BuffBonusResultPopup>()
                .OnBuffSelectedSubject
                .Subscribe(_ => SetIcon(skillState))
```

**Review by sonohoshi — APPROVED** [review]

**Review by boscohyun — APPROVED** [review]

**Pull request merged**